zuul-core/src/main/java/com/netflix/netty/common/ByteBufUtil.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.netty.common;
+
+import static io.netty.util.ResourceLeakDetector.Level.ADVANCED;
+import static io.netty.util.ResourceLeakDetector.Level.PARANOID;
+import com.netflix.zuul.message.ZuulMessage;
+import io.netty.util.ReferenceCounted;
+import io.netty.util.ResourceLeakDetector;
+
+/**
+ * ByteBufUtil
+ *
+ * @author Arthur Gonigberg
+ * @since October 20, 2022
+ */
+public class ByteBufUtil {
+
+    private static final boolean isAdvancedLeakDetection =
+            ResourceLeakDetector.getLevel() == PARANOID || ResourceLeakDetector.getLevel() == ADVANCED;
+
+    public static void touch(ReferenceCounted byteBuf, String hint, ZuulMessage msg) {
```

Comment:
nit: could use String.format with vargs to allow for future logging of other objects if required

Reply:
I was thinking about that too, but String.format has terrible performance.